addOns/llm/src/main/java/org/zaproxy/addon/llm/services/LlmCommunicationService.java

```diff
@@ -54,116 +66,123 @@ public class LlmCommunicationService {
 
             // Use the existing importSwagger method
-            endpoint_count = importHttpCalls(openApiDefinition);
+            endpointCount = importHttpCalls(openApiDefinition);
 
         } catch (Exception e) {
-            e.printStackTrace();
+            LOGGER.error(e.getMessage());
         }
-        return endpoint_count;
+        return endpointCount;
     }
 
     public void reviewAlert(Alert alert) {
 
-        Alert updated_alert = alert;
-        Alert original_alert = updated_alert.newInstance();
+        Alert updatedAlert = alert;
+        Alert originalAlert = updatedAlert.newInstance();
 
-        if(! alert.getTags().containsKey(AI_REVIEWD_TAG_KEY)) {
+        if (!alert.getTags().containsKey(AI_REVIEWD_TAG_KEY)) {
             Confidence conf_llm;
             LOGGER.debug("Reviewing alert :" + alert.getName());
             LOGGER.debug("Confidence level from ZAP : " + alert.getConfidence());
             conf_llm = llmAssistant.review(alert.getDescription(), alert.getEvidence());
-            LOGGER.debug("Confidence level from LLM : " + conf_llm.getLevel() + " | Explanation : " + conf_llm.getExplanation());
-            updated_alert.setConfidence(conf_llm.getLevel());
-            updated_alert.setOtherInfo("LLM Explanation : " + conf_llm.getExplanation() + "\n" + alert.getOtherInfo());
+            LOGGER.debug(
+                    "Confidence level from LLM : "
+                            + conf_llm.getLevel()
+                            + " | Explanation : "
+                            + conf_llm.getExplanation());
+            updatedAlert.setConfidence(conf_llm.getLevel());
+            updatedAlert.setOtherInfo(
+                    "LLM Explanation : " + conf_llm.getExplanation() + "\n" + alert.getOtherInfo());
             Map<String, String> alertTags = alert.getTags();
 
-            alertTags.putIfAbsent(AI_REVIEWD_TAG_KEY,AI_REVIEWD_TAG_VALUE);
-            updated_alert.setTags(alertTags);
+            alertTags.putIfAbsent(AI_REVIEWD_TAG_KEY, AI_REVIEWD_TAG_VALUE);
+            updatedAlert.setTags(alertTags);
 
             try {
-                getExtAlert().updateAlert(updated_alert);
-                getExtAlert().updateAlertInTree(original_alert, updated_alert);
+                getExtAlert().updateAlert(updatedAlert);
+                getExtAlert().updateAlertInTree(originalAlert, updatedAlert);
                 if (alert.getHistoryRef() != null) {
-                    alert.getHistoryRef().updateAlert(updated_alert);
+                    alert.getHistoryRef().updateAlert(updatedAlert);
                     if (alert.getHistoryRef().getSiteNode() != null) {
                         // Needed if the same alert was raised on another href for the same
                         // SiteNode
-                        alert.getHistoryRef().getSiteNode().updateAlert(updated_alert);
+                        alert.getHistoryRef().getSiteNode().updateAlert(updatedAlert);
```

Comment:
I'm confused about these conditionals and doing the same action over and over, shouldn't the call hat 167 have done it regardless of the conditions?

Reply:
I don't why but without this I get this exception for some alerts : 

```
Index -1 out of bounds for length 10
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 10
```

I used the same code from this extension : https://github.com/zaproxy/zap-extensions/blob/main/addOns/alertFilters/src/main/java/org/zaproxy/zap/extension/alertFilters/ExtensionAlertFilters.java#L459

Reply:
This is the "normal" loop we use for reading. I guess part of the problem you're encountering might be timing with the updates. I'd have to play with it and see if your conditions can be simplified or worked around somehow.

https://github.com/zaproxy/community-scripts/blob/main/standalone/Loop%20through%20alerts.js